src/main/java/com/mraof/minestuck/data/BetterCombatProvider.java

```diff
@@ -0,0 +1,183 @@
+package com.mraof.minestuck.data;
+
+import com.google.gson.*;
+import com.mraof.minestuck.Minestuck;
+import com.mraof.minestuck.item.MSItems;
+import com.mraof.minestuck.item.weapon.MSToolType;
+import com.mraof.minestuck.item.weapon.WeaponItem;
+import net.minecraft.MethodsReturnNonnullByDefault;
+import net.minecraft.data.CachedOutput;
+import net.minecraft.data.DataProvider;
+import net.minecraft.data.PackOutput;
+import net.minecraft.world.item.Item;
+import net.neoforged.neoforge.registries.DeferredItem;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+import java.nio.file.Path;
+import java.util.*;
+import java.util.concurrent.CompletableFuture;
+
+import static com.mraof.minestuck.item.MSItemTypes.*;
+
+@ParametersAreNonnullByDefault
+@MethodsReturnNonnullByDefault
+public class BetterCombatProvider implements DataProvider
+{
+	private final PackOutput output;
+	
+	Map<String, String> weaponWithParent = new HashMap<>();
```

Comment:
I recommend changing the key type from `String` to `ResourceKey<Item>` to better reflect what the key is supposed to be.